tests/baselines/reference/objectRest.errors.txt

```diff
@@ -56,7 +59,12 @@ tests/cases/conformance/types/rest/objectRest.ts(43,35): error TS2537: Type '{ a
 !!! error TS2537: Type '{ a: number; b: string; }' has no matching index signature for type 'string'.
                                       ~~~~~~~~~
 !!! error TS2537: Type '{ a: number; b: string; }' has no matching index signature for type 'string'.
+                                                            ~
+!!! error TS2403: Subsequent variable declarations must have the same type.  Variable 'o' must be of type '{ a: number; b: string; }', but here has type '{}'.

```

Comment:
I'm not sure what's going on here. The existing errors indicate that rest *won't* remove 'a' and 'b' because they actually have type string, but the new error makes it look like the resulting type is `{}`, indicating that the rest *did* remove 'a' and 'b'.

Probably the place to start is to declare a new variable instead of trying to redeclare `o`. I don't know if that was intentional in the original test.

Reply:
Previously we just got computed properties wrong. We effectively ignored them when computing the rest type. Now we actually factor them in, and when a computed property has type `string` (i.e. a non-unit type) we remove _all_ properties from the rest type because we have no idea which ones were actually removed. This mirrors what `Pick<T, Exclude<keyof T, string>>` does.